lib/utils/errors.ts

```diff
@@ -50,8 +54,16 @@ export const of = (error: any): RuntimeError =>
         error instanceof Error ? error : undefined
       );
 
-export type RuntimeError = TransientError | PermanentError | UnknowError;
+export type RuntimeError =
+  | TransientError
+  | PermanentError
+  | UnknowError
+  | ExpiredError;
 
 // tslint:disable-next-line:no-any
 export const isTransient = (error: any): error is TransientError =>
   error.kind && error.kind === ErrorTypes.TransientError;
+
+// tslint:disable-next-line:no-any
+export const isExpired = (error: any): error is ExpiredError =>
```

Comment:
why `any` here? can't it be restricted to `RuntimeError`? (also `isTransient`)

also the fact that there's `ErrorTypes.ExpiredError`, `type ExpiredError`, `const ExpiredError`, `const RuntimeError` and `type RuntimeError` looks very confusing and overly complicated

Reply:
right, I'll try to refactor this using io-ts (we need a hiearchy of types which can be discovered at runtime)
but we'll still got const and type (at least)

Reply:
even using io-ts the situation does not change, see https://gist.github.com/gunzip/ede14c7fd913b913222dd1f109cbda86

what do you suggest to simplify it ?